bus/bus.go

```diff
@@ -641,13 +658,14 @@ func New(s Syncer, cm ChainManager, tp TransactionPool, w Wallet, hdb HostDB, cs
 		"GET    /hosts/:hostkey": b.hostsPubkeyHandlerGET,
 		"POST   /hosts/:hostkey": b.hostsPubkeyHandlerPOST,
 
-		"GET    /contracts":             b.contractsHandler,
-		"GET    /contracts/:id":         b.contractsIDHandlerGET,
-		"POST   /contracts/:id/new":     b.contractsIDNewHandlerPOST,
-		"POST   /contracts/:id/renewed": b.contractsIDRenewedHandlerPOST,
-		"DELETE /contracts/:id":         b.contractsIDHandlerDELETE,
-		"POST   /contracts/:id/acquire": b.contractsAcquireHandler,
-		"POST   /contracts/:id/release": b.contractsReleaseHandler,
+		"GET    /contracts":               b.contractsHandler,
+		"GET    /contracts/:id":           b.contractsIDHandlerGET,
+		"GET    /contracts/:id/ancestors": b.contractsAncestorsHandlerGet,
+		"POST   /contracts/:id/new":       b.contractsIDNewHandlerPOST,
```

Comment:
could this be `PUT /contracts/:id`?

Reply:
Is `POST /contracts/:id` fine? Because we want the created contract to be returned.

Reply:
ah, I see -- `PUT` would be inappropriate anyway because we're not sending a `bus.Contract`.

I think `POST /contracts`  might be best. Doesn't really make sense to have the ID in the route before the resource exists.

(What I *really* think is that this REST stuff seems to generate a lot of ultimately-pointless bikesheding about verbs and routes and query params vs request bodies etc. etc. etc.; maybe the next time I have to design an API it will just have a single `POST /rpc` endpoint...)